src/__tests__/components/Comment.test.js

```diff
@@ -0,0 +1,224 @@
+  articleSlug: '',
+  commentList: [],
+  currentUser: {},
+  onDeleteComment: jest.fn(),
+  onUpdateComment: jest.fn(),
+  updateBody: '',
+  onUpdateCommentInput: jest.fn(),
+  isLoggedIn: false,
+  history: {
+    push: jest.fn(),
+  },
+  onLikeComment: jest.fn(),
+  onDislikeComment: jest.fn(),
+};
+const mockStore = configureMockStore([thunk]);
+const store = mockStore(initialState);
+
+describe('<Comment />', () => {
+  test('should render comment component', () => {
+    const comment = shallow(<Comment {...props} />);
+    expect(comment).toMatchSnapshot();
+  });
+
+  test('should display comments', () => {
+    const newProps = { ...props, commentList: commentData };
+    const wrapper = mount(
+      <Provider store={store}>
+        <Comment {...newProps} />
+      </Provider>,
+    );
+    const comment = wrapper.find('Comment');
+    const { displayComments } = comment.instance();
+    const comments = displayComments();
+    expect(comments.length).toEqual(3);
+  });
+
+  test('should update comment', () => {
+    const fakeId = 'fake-id';
+    const wrapper = mount(
+      <Provider store={store}>
+        <Comment {...props} />
+      </Provider>,
+    );
+    const comment = wrapper.find('Comment');
+    const { onEditComment } = comment.instance();
+    onEditComment(fakeId);
+    expect(comment.props().onUpdateComment).toHaveBeenCalled();
+  });
+
+  test('should add comment', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <Comment {...props} />
+      </Provider>,
+    );
+    const comment = wrapper.find('Comment');
+    const { addComment } = comment.instance();
+    addComment();
+    expect(comment.props().onSubmitComment).toHaveBeenCalled();
+  });
+
+  test('should update input as user type', () => {
+    const event = {
+      preventDefault: jest.fn(),
+      target: {
+        value: ' ',
+      },
+    };
+    const wrapper = mount(
+      <Provider store={store}>
+        <Comment {...props} />
+      </Provider>,
+    );
+    const comment = wrapper.find('Comment');
+    const { onChange } = comment.instance();
+    onChange(event);
+    expect(comment.props().onCommentInput).toHaveBeenCalled();
+    expect(event.preventDefault).toHaveBeenCalled();
+  });
+
+  test('should update input as user type', () => {
+    const event = {
+      preventDefault: jest.fn(),
+      target: {
+        value: ' ',
+      },
+    };
+    const wrapper = mount(
+      <Provider store={store}>
+        <Comment {...props} />
+      </Provider>,
+    );
+    const comment = wrapper.find('Comment');
+    const mockFn = jest.spyOn(comment.instance(), 'onEnterPress');
+    const commentInput = comment.find('textarea');
+    commentInput.simulate('keydown', event);
+    expect(mockFn).toHaveBeenCalled();
+  });
+
+  describe('onEnterPress', () => {
+    const event = {
+      keyCode: 13,
+      shiftKey: false,
+      preventDefault: jest.fn(),
+      target: {
+        value: ' ',
```

Comment:
is `value` an empty string or `space`? There's a difference

Reply:
This is a space because we wanted to test trimming.